src/webnn_native/ops/Binary.cpp

```diff
@@ -19,6 +19,76 @@
 
 namespace webnn_native { namespace op {
 
+    MaybeError Binary::CalculateShape() {
+        auto inputShape1 =
+            mInputs[0]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[0]->Shape();
+        auto inputShape2 =
+            mInputs[1]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[1]->Shape();
+
+        auto l1 = inputShape1.size(), l2 = inputShape2.size();
+        bool shape1IsBigger = l1 >= l2;
+        auto maxShape = shape1IsBigger ? inputShape1 : inputShape2;
+        auto minShape = shape1IsBigger ? inputShape2 : inputShape1;
```

Comment:
Why do you need to define `maxShape` and `minShape`? I suppose you only need to find the max rank between rankA and rankB.

spec says:
> The rank of the output tensor is the maximum rank of the input tensors. For each dimension of the output tensor, its size is the maximum size along that dimension of the input tensors.

Reply:
I need the maxShape as the default value of broadcasted shape, refer to https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc#L299. Does it make sense? Thanks! @huningxin